std/regex/package.d

```diff
@@ -340,8 +342,13 @@ public alias Regex(Char) = std.regex.internal.ir.Regex!(Char);
 
     No longer used, kept as alias to Regex for backwards compatibility.
 +/
+// See commentary at https://github.com/dlang/phobos/pull/5963
+deprecated(`use typeof(ctRegex!("foo"))`)
 public alias StaticRegex = Regex;
 
+private alias _Regex(Char) = std.regex.internal.ir.Regex!(Char);
```

Comment:
Ugh!

Reply:
Btw that's another advantage of using `auto`:
We don't have to pay any initialization cost until the function is called once :)

Reply:
You mean import cost, no? Because "initializers" would be constructors, usually. You never pay for an instance initializer until an instance is created, and you always pay for a static constructor.

Anyway, that's easily addressed:

```D
template _Regex(Char)
{
  private import std.regex.internal.ir : IRRegex = Regex;
  alias _Regex = IRRegex!Char;
}
```

Pretty sure this technique has been discussed on the newsgroup several times.

Reply:
I meant CTFE "initialization" cost - evaluation would have been a better word.

> Pretty sure this technique has been discussed on the newsgroup several times.

Well NG and actual PRs is a huge difference. Here's an overview on the status quo on making Phobos slicker:
- https://github.com/dlang/phobos/pull/5942
- https://github.com/dlang/phobos/pull/5931
- https://github.com/dlang/phobos/pull/5945
- https://github.com/dlang/phobos/pull/5948
- https://github.com/dlang/phobos/pull/5916
- http://forum.dlang.org/post/zezhvfdwutpdqkbkjymp@forum.dlang.org (latest NG thread)

tl;dr: there are still a lot of easy fruits to pick :)